include/swift/AST/Attr.h

```diff
@@ -134,8 +148,17 @@ class TypeAttributes {
     return true;
   }
   
-  bool hasConvention() const { return convention.hasValue(); }
-  StringRef getConvention() const { return *convention; }
+  bool hasConvention() const { return ConventionOf.hasValue(); }
+
+  /// Returns the primary calling convention string.
+  ///
+  /// Note: For C conventions, this may not represent the full convention.
+  StringRef getConvention() const { return ConventionOf.getValue().Name; }
+
+  /// Show the string enclosed between @convention(..)'s parentheses.
+  ///
+  /// For example, @convention(foo, bar) will give the string "foo, bar".
+  void getFullConventionString(SmallVectorImpl<char> &buffer) const;
```

Comment:
`getConventionArguments()`, maybe?  Let's not call this "full" when it doesn't include the `@convention` part.